### PR TITLE
wrapper emit result as base64 string

### PR DESCRIPTION
**Pull request by Elli610**

### CODE REVIEW

File: packages/as-transformer/src/transformers/massaExport.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export class MassaExport {
       wrapper +=
         '  generateEvent(' +
         `\`Result${this.functionName}:` +
-        "'${response.buffer.toString()}'`);\n";
+        "'${byteArrayToBase64(new Uint8Array(response.buffer))}'`);\n";
```
Comment (gregLibert) [inline packages/as-transformer/src/transformers/massaExport.ts:149]: Why not directly using response which is an Uint8array?

File: packages/as-types/assembly/uint8ArrayToBase64.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Converts a Uint8Array to a base64 string.
+ *
+ * @param bytes - The Uint8Array to convert.
+ *
+ * @returns The base64 string.
+ */
+export function uint8ArrayToBase64(bytes: Uint8Array): string {
```
Comment (gregLibert) [inline packages/as-types/assembly/uint8ArrayToBase64.ts:8]: why not using https://github.com/near/as-base64 instead ?

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/uint8ArrayToBase64.ts (hunk 2 above)
Comment (Elli610) [reply packages/as-types/assembly/uint8ArrayToBase64.ts:8]: I didn't know about this library

File: packages/as-transformer/src/transformers/massaExport.ts (hunk 1 above)
Comment (Elli610) [reply packages/as-transformer/src/transformers/massaExport.ts]: idk

File: packages/as-transformer/src/transformers/massaExport.ts [hunk 3]
```diff
@@ -146,7 +146,7 @@ export class MassaExport {
       wrapper +=
         '  generateEvent(' +
         `\`Result${this.functionName}:` +
-        "'${response.buffer.toString()}'`);\n";
+        "'${new Uint8Array(response.buffer).toString()}'`);\n";
```
Comment (gregLibert) [inline packages/as-transformer/src/transformers/massaExport.ts:149]:
```suggestion
        "'${encode(response)}'`);\n";
```
with the following import:
```typescript
import { encode } from "as-base64";
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/as-transformer/src/transformers/massaExport.ts [hunk 4]
```diff
@@ -319,6 +319,14 @@ export class MassaExport {
       );
     }
 
+    // checking if 'import { encode } from "as-base64";' is needed or if they are already imported by the contract
+    const encodeImportRegex =
```
Comment (gregLibert) [inline packages/as-transformer/src/transformers/massaExport.ts:323]:
```suggestion
    const encodeBase64ImportRegex =
```

File: packages/as-transformer/src/transformers/massaExport.ts [hunk 5]
```diff
@@ -319,6 +319,14 @@ export class MassaExport {
...
+      /(?:import\s*{.*encode.*}\s*from\s*("|')as-base64("|'))/gm;
+
+    if (encodeImportRegex.exec(content) === null) {
```
Comment (gregLibert) [inline packages/as-transformer/src/transformers/massaExport.ts:326]:
```suggestion
    if (encodeBase64ImportRegex.exec(content) === null) {
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**